lib/post_creator.rb

```diff
@@ -161,7 +161,7 @@ def valid?
       return false
     end
 
-    DiscourseEvent.trigger :before_create_post, @post
+    DiscourseEvent.trigger :before_create_post, @post, @opts
```

Comment:
Can we add a test for this and for `before_edit_post`? (These two changes could also be moved to a separate PR, if you would like.)

Reply:
handled in ad960d66b6